cli/cli.go

```diff
@@ -49,6 +49,12 @@ var (
 	LatestReleaseURL = "https://api.github.com/repos/exercism/cli/releases/latest"
 )
 
+// Updater is a simple upgradable file interface.
+type Updater interface {
+	IsUpToDate() (bool, error)
+	Upgrade() error
+}
```

Comment:
Thanks! My appreciation for interfaces and embedding went through the roof as I was writing this.